### PR TITLE
[Merged by Bors] - feat(Probability/Distributions): formalize Pareto distribution

### PR DESCRIPTION
**Pull request by arulandu**
Formalize Pareto distribution, using Gamma distribution as a reference.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (arulandu) [reply Mathlib/Probability/Distributions/Pareto.lean:75]: bot fix style

**Comment by arulandu** [comment]
#16713

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:110]: A bit shorter:
```suggestion
  · rw [integral_Ici_eq_integral_Ioi, integral_mul_left, integral_Ioi_rpow_of_lt _ ht]
    · field_simp [hr]
      rw [mul_assoc, ← rpow_add ht]
      simp
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:120]:
```suggestion
    simp_rw [paretoPDFReal, eq_true_intro hx, ite_true]
 
end ParetoPDF
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:64]: This doesn't hurt
```suggestion
@[measurability, fun_prop]
lemma measurable_paretoPDFReal (t r : ℝ) : Measurable (paretoPDFReal t r) :=
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:69]:
```suggestion
@[measurability, fun_prop]
 lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:35]: Just a detail but I think this is more common, same below
```suggestion
noncomputable def paretoPDFReal (t r x : ℝ) : ℝ :=
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:145]: Once you changed the hypothesis of `paretoPDFReal_nonneg`, you can do
```suggestion
  · exact ae_of_all _ fun b ↦ by simp only [Pi.zero_apply, paretoPDFReal_nonneg ht.le hr.le]
```

**Review by EtienneC30 — COMMENTED** [review]
That's a great job! My comments are mostly small details.

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (arulandu) [reply Mathlib/Probability/Distributions/Pareto.lean:69]: I get a "unrecognized function property" when I try to do this here. the previous occurrence works though

**Comment by arulandu** [comment]
@EtienneC30 Fixed!

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [reply Mathlib/Probability/Distributions/Pareto.lean:67]: Ah, this will be because strong measurability is not conisdered as something that should be proved with `fun_prop` leave it as it is then.

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:148]: The double `simp only` is not very satisfactory, this is nicer I think.
```suggestion
lemma paretoCDFReal_eq_lintegral {t r : ℝ} (ht : 0 < t) (hr : 0 < r) (x : ℝ) :
    paretoCDFReal t r x = ENNReal.toReal (∫⁻ x in Iic x, paretoPDF t r x) := by
  have : IsProbabilityMeasure (paretoMeasure t r) := isProbabilityMeasurePareto ht hr
  rw [paretoCDFReal, cdf_eq_toReal, paretoMeasure, withDensity_apply _ measurableSet_Iic]
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:17]:
```suggestion
  scale `t` and shape `r` (when `ht : 0 < t` and `hr : 0 < r`).
```

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:113]: Please avoid multiline tactic blocks in the arguments of a theorem. Use `apply` or `refine` instead.
```suggestion
   refine ae_of_all _ fun x (hx : t ≤ x) ↦ ?_
    have := lt_of_lt_of_le ht hx
    positivity
```

**Review by EtienneC30 — COMMENTED** [review]
A few more details in the doc, and some advice on the proof style.

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:127]:
```suggestion
lemma isProbabilityMeasure_paretoMeasure {t r : ℝ} (ht : 0 < t) (hr : 0 < r) :
    IsProbabilityMeasure (paretoMeasure t r) where
```

**Comment by arulandu** [comment]
@EtienneC30 fixed again!

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:69]: Bad indent
```suggestion
@[measurability]
lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
    StronglyMeasurable (paretoPDFReal t r) :=
  (measurable_paretoPDFReal t r).stronglyMeasurable
```

Comment (EtienneC30) [inline Mathlib/Probability/Distributions/Pareto.lean:69]: Bad indent
```suggestion
@[measurability]
lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
    StronglyMeasurable (paretoPDFReal t r) :=
  (measurable_paretoPDFReal t r).stronglyMeasurable
```

**Review by EtienneC30 — APPROVED** [review]
This seems all right to me! Now we need a maintainer to allow merging.

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Probability/Distributions/Pareto.lean:33]:
```suggestion
/-- The pdf of the Pareto distribution depending on its scale `t` and rate `r`. -/
```
(assuming I got those correct).

**Review by RemyDegenne — COMMENTED** [review]
It looks good! Only two final comments from me, and then we can merge this.

File: Mathlib/Probability/Distributions/Pareto.lean
Comment (arulandu) [reply Mathlib/Probability/Distributions/Pareto.lean:44]: Makes sense, fixing

**Comment by arulandu** [comment]
@RemyDegenne Fixed! Are there any other changes I should add before we can merge this?

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**
